### ISSUE TITLE
ERROR: unable to download video data: [Errno 2] No such file or directory, When using aria2 as external downloader

### ISSUE THREAD

**Issue by tzl2019** [yt-dlp/yt-dlp#7188]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2023.03.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

When using aria2 as downloader to get videos on bilibili,it is normal to download a video that contains just one P. But error occurs when one url contains several Ps.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-f', 'ba', '--cookies-from-browser', 'chrome', 'https://www.bilibili.com/video/BV1bu411p7Jf/', '--downloader', 'aria2c', '--downloader-args', "--log log.txt -D", '-o', '%(id)s.%(ext)s', '-vU']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (pip)
[debug] Python 3.11.3 (CPython x86_64 64bit) - macOS-10.16-x86_64-i386-64bit (OpenSSL 1.1.1t  7 Feb 2023)
[debug] exe versions: ffmpeg 6.0 (setts), ffprobe 6.0
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2023.05.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-11.0.3
[Cookies] Extracting cookies from chrome
[debug] using find-generic-password to obtain password from OSX keychain
[Cookies] Extracted 1137 cookies from chrome
[debug] cookie version breakdown: {'v10': 1125, 'other': 0, 'unencrypted': 12}
[debug] Loaded 1786 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
yt-dlp is up to date (stable@2023.03.04)
[BiliBili] Extracting URL: https://www.bilibili.com/video/BV1bu411p7Jf/
[BiliBili] 1bu411p7Jf: Downloading webpage
[BiliBili] BV1bu411p7Jf: Extracting videos in anthology
[BiliBili] Format(s) 1080P 高码率 are missing; you have to login or become premium member to download them. Use --cookies-from-browser or --cookies for the authentication. See  https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp  for how to manually pass cookies
[BiliBili] 529297068: Extracting chapters
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] BV1bu411p7Jf: Downloading 1 format(s): 2
[debug] Invoking aria2c downloader on "https://upos-hz-mirrorakam.akamaized.net/upgcxcode/21/43/1147734321/1147734321-1-30280.m4s?e=ig8euxZM2rNcNbdlhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1685609195&gen=playurlv2&os=akam&oi=794458175&trid=1cab70dc68ed436fbae9b53c8e869417u&mid=97058635&platform=pc&upsig=20c5f5cf0252d2a0aa80d76d174a1e04&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,mid,platform&hdnts=exp=1685609195~hmac=a8b2d9d8eea4800cf54561f9e5cc2919567ad494fd9617c86525e7f1efd26838&bvc=vod&nettype=0&orderid=0,1&buvid=9D9A7151-CEA0-5890-0C31-A782A8296DA574754infoc&build=0&agrr=1&bw=23602&logo=80000000"
[download] Destination: BV1bu411p7Jf.m4a
[debug] aria2c command line: aria2c -c --console-log-level=warn --summary-interval=0 --download-result=hide --http-accept-gzip=true --file-allocation=none -x16 -j16 -s16 --min-split-size 1M --header 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8' --header 'Accept-Language: en-us,en;q=0.5' --header 'Sec-Fetch-Mode: navigate' --header 'Referer: https://www.bilibili.com/video/BV1bu411p7Jf/'  --check-certificate=true --remote-time=true --show-console-readout=true --conf-path=/Users/tangzhili/.aria2/aria2.conf --log log.txt -D --out ./BV1bu411p7Jf.m4a.part --auto-file-renaming=false -- 'https://upos-hz-mirrorakam.akamaized.net/upgcxcode/21/43/1147734321/1147734321-1-30280.m4s?e=ig8euxZM2rNcNbdlhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1685609195&gen=playurlv2&os=akam&oi=794458175&trid=1cab70dc68ed436fbae9b53c8e869417u&mid=97058635&platform=pc&upsig=20c5f5cf0252d2a0aa80d76d174a1e04&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,mid,platform&hdnts=exp=1685609195~hmac=a8b2d9d8eea4800cf54561f9e5cc2919567ad494fd9617c86525e7f1efd26838&bvc=vod&nettype=0&orderid=0,1&buvid=9D9A7151-CEA0-5890-0C31-A782A8296DA574754infoc&build=0&agrr=1&bw=23602&logo=80000000'
ERROR: unable to download video data: [Errno 2] No such file or directory: 'BV1bu411p7Jf.m4a.part'
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 3247, in process_info
    success, real_download = self.dl(temp_filename, info_dict)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 2970, in dl
    return fd.download(name, new_info, subtitle)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/yt_dlp/downloader/common.py", line 444, in download
    ret = self.real_download(filename, info_dict)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/yt_dlp/downloader/external.py", line 66, in real_download
    fsize = os.path.getsize(encodeFilename(tmpfilename))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen genericpath>", line 50, in getsize
FileNotFoundError: [Errno 2] No such file or directory: 'BV1bu411p7Jf.m4a.part'
```

**Comment by bashonly** [comment]
> Copy the **WHOLE** output (starting with `[debug] Command-line config`) and insert it below

**Comment by tzl2019** [comment]
This is the whloe output info:
[debug] Command-line config: ['-f', 'ba', '--proxy', 'http://bihbj.zhenguanyu.com:8118', '--cookies-from-browser', 'chrome', 'https://www.bilibili.com/video/BV12g411B7LF/', '--downloader', 'aria2c', '--downloader-args', "--conf-path=/Users/tangzhili/.aria2/aria2.conf -x 16 -s 16 -k 1M --log 'log2.txt' -D", '-o', '%(id)s.%(ext)s', '-vU']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (pip)
[debug] Python 3.11.3 (CPython x86_64 64bit) - macOS-10.16-x86_64-i386-64bit (OpenSSL 1.1.1t  7 Feb 2023)
[debug] exe versions: ffmpeg 6.0 (setts), ffprobe 6.0
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2023.05.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-11.0.3
[Cookies] Extracting cookies from chrome
[debug] Extracting cookies from: "/Users/tangzhili/Library/Application Support/Google/Chrome/Default/Cookies"
[debug] using find-generic-password to obtain password from OSX keychain
[Cookies] Extracted 1125 cookies from chrome
[debug] cookie version breakdown: {'v10': 1113, 'other': 0, 'unencrypted': 12}
[debug] Proxy map: {'http': 'http://bihbj.zhenguanyu.com:8118', 'https': 'http://bihbj.zhenguanyu.com:8118'}
WARNING: External Downloader arguments given without specifying name. The arguments will be given to aria2c
[debug] Loaded 1786 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
yt-dlp is up to date (stable@2023.03.04)
[BiliBili] Extracting URL: https://www.bilibili.com/video/BV12g411B7LF/
[BiliBili] 12g411B7LF: Downloading webpage
[BiliBili] BV12g411B7LF: Extracting videos in anthology
[BiliBili] Downloading playlist BV12g411B7LF - add --no-playlist to download just the video BV12g411B7LF
[download] Downloading playlist: 【有声书】司马迁《史记》原文精讲 完整版
[BiliBili] Playlist 【有声书】司马迁《史记》原文精讲 完整版: Downloading 100 items of 100
[download] Downloading item 1 of 100
[BiliBili] Extracting URL: https://www.bilibili.com/video/BV12g411B7LF?p=1
[BiliBili] 12g411B7LF: Downloading webpage
[BiliBili] BV12g411B7LF: Extracting videos in anthology
[BiliBili] 517348089: Extracting chapters
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] BV12g411B7LF_p1: Downloading 1 format(s): 2
[debug] Invoking aria2c downloader on "https://upos-hz-mirrorakam.akamaized.net/upgcxcode/30/84/882328430/882328430_nb3-1-30280.m4s?e=ig8euxZM2rNcNbdlhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1685598318&gen=playurlv2&os=akam&oi=794458175&trid=4ff75faf54554197bee0e03a5811ff60u&mid=97058635&platform=pc&upsig=16f02dd96cd897ee6f490ed4f5bddbc8&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,mid,platform&hdnts=exp=1685598318~hmac=b52f2aa00e01f58d686fa0030e2d6051baa266209397539f5e52de8eb616ec07&bvc=vod&nettype=0&orderid=0,1&buvid=9D9A7151-CEA0-5890-0C31-A782A8296DA574754infoc&build=0&agrr=1&bw=23887&logo=80000000"
[download] Destination: BV12g411B7LF_p1.m4a
[debug] aria2c command line: aria2c -c --console-log-level=warn --summary-interval=0 --download-result=hide --http-accept-gzip=true --file-allocation=none -x16 -j16 -s16 --min-split-size 1M --header 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.115 Safari/537.36' --header 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8' --header 'Accept-Language: en-us,en;q=0.5' --header 'Sec-Fetch-Mode: navigate' --header 'Referer: https://www.bilibili.com/video/BV12g411B7LF?p=1' --all-proxy http://bihbj.zhenguanyu.com:8118 --check-certificate=true --remote-time=true --show-console-readout=true --conf-path=/Users/tangzhili/.aria2/aria2.conf -x 16 -s 16 -k 1M --log log2.txt -D --out ./BV12g411B7LF_p1.m4a.part --auto-file-renaming=false -- 'https://upos-hz-mirrorakam.akamaized.net/upgcxcode/30/84/882328430/882328430_nb3-1-30280.m4s?e=ig8euxZM2rNcNbdlhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1685598318&gen=playurlv2&os=akam&oi=794458175&trid=4ff75faf54554197bee0e03a5811ff60u&mid=97058635&platform=pc&upsig=16f02dd96cd897ee6f490ed4f5bddbc8&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,mid,platform&hdnts=exp=1685598318~hmac=b52f2aa00e01f58d686fa0030e2d6051baa266209397539f5e52de8eb616ec07&bvc=vod&nettype=0&orderid=0,1&buvid=9D9A7151-CEA0-5890-0C31-A782A8296DA574754infoc&build=0&agrr=1&bw=23887&logo=80000000'
ERROR: unable to download video data: [Errno 2] No such file or directory: 'BV12g411B7LF_p1.m4a.part'
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 3247, in process_info
    success, real_download = self.dl(temp_filename, info_dict)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 2970, in dl
    return fd.download(name, new_info, subtitle)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/yt_dlp/downloader/common.py", line 444, in download
    ret = self.real_download(filename, info_dict)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/yt_dlp/downloader/external.py", line 66, in real_download
    fsize = os.path.getsize(encodeFilename(tmpfilename))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen genericpath>", line 50, in getsize
FileNotFoundError: [Errno 2] No such file or directory: 'BV12g411B7LF_p1.m4a.part'

**Comment by bashonly** [comment]
remove `-D` from your downloader-args

from aria2c docs:
```
       -D, --daemon [true|false]
              Run as daemon. The current working directory will be changed  to
              / and standard input, standard output and standard error will be
              redirected to /dev/null. Default: false
```
